recipes/bzip2/all/test_package/conanfile.py

```diff
@@ -1,17 +1,25 @@
 import os
-from conans import ConanFile, CMake, tools
+from conans import ConanFile, tools
+from conan.tools.cmake import CMake
+from conan.tools.layout import cmake_layout
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "arch", "build_type"
-    generators = "cmake", "cmake_find_package"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
 
     def build(self):
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
+    def configure(self):
+        del self.settings.compiler.libcxx
+
+    def layout(self):
```

Comment:
seems to be unused (e.g. never called), it is needed? if so, why?

Reply:
It is the new ``layout()`` built-in Conan method to define the package layout, it is called internally by Conan, like other methods: https://docs.conan.io/en/latest/reference/conanfile/methods.html#layout

Reply:
reading about feature, I don't get why is it needed for test package?

Reply:
It is not "needed" but nice to use always the same layout, more clear, and easy to explore the build files in case you need.